client/js/compute.js

```diff
@@ -0,0 +1,578 @@
+      sample: 'var a = from ${x}',
+      call: function(v, x) { return v = Array.from(x) }
+    },
+    {
+      name: 'isArray',
+      desc: 'returns true if the object is an array, and false if not',
+      sample: 'var a = isArray ${x}',
+      call: function(v, x) { return v = Array.isArray(x) }
+    },
+    {
+      name: 'concatArray',
+      desc: 'returns an array containing the elements of the two joined arrays',
+      sample: 'var a = ${x} concatArray ${y}',
+      call: function(v, x, y) { return v = x.concat(y) }
+    },
+    {
+      name: 'pop',
+      desc: 'returns an array after removing its last element',
+      sample: 'var a = ${x} pop',
+      call: function(v, x) { return v = x.pop() }
+    },
+    {
+      name: 'reverse',
+      desc: 'returns an array after reversing the order of its elements',
+      sample: 'var a = ${x} reverse',
+      call: function(v, x) { return v = x.reverse() }
+    },
+    {
+      name: 'shift',
+      desc: 'returns an array after removing its first element',
+      sample: 'var a = ${x} shift',
+      call: function(v, x) { return v = x.shift() }
+    },
+    {
+      name: 'sort',
+      desc: 'returns an array after sorting its elements alphabetically',
+      sample: 'var a = ${x} sort',
+      call: function(v, x) { return v = x.sort() }
+    },
+    {
+      name: 'numericSort',
+      desc: 'returns an array after sorting its elements numerically',
+      sample: 'var a = ${x} numericSort',
+      call: function(v, x) { return v = x.sort((a, b) => a.toString().localeCompare(b, undefined, {numeric: true, ignorePunctuation: true})) }
+    },
+    {
+      name: 'findIndex',
+      desc: 'returns the index of the first element in the array that matches the given value y',
+      sample: 'var a = ${x} findIndex ${y}',
+      call: function(v, x, y) { return v = x.findIndex(y) }
+    },
+    {
+      name: 'join',
+      desc: 'returns the array x as a combined string with values separated by an optional separator y, defaulting to comma',
+      sample: 'var a = ${x} join ${y}',
+      call: function(v, x, y) { return v = (y !== undefined) ? x.join(y) : x.join() }
+    },
+    {
+      name: 'slice',
+      desc: 'returns elements between start index y an end index z of an array (or characters of a string), excluding the end (unless z is omitted)',
+      sample: 'var a = ${x} slice ${y} ${z}',
+      call: function(v, x, y, z) { return v = (z !== undefined) ? x.slice(y, z) : x.slice(y) }
+    },
+    {
+      name: 'splice',
+      desc: 'removes elements starting from index x from the array in variable a (optionally limited to only y elements)',
+      sample: 'var a = splice ${x} ${y}',
+      call: function(v, x, y) { if(y !== undefined) {v.splice(x, y)} else {v.splice(x)}; return v }
+    },
+    {
+      name: 'push',
+      desc: 'adds the given value as the last array element of the variable',
+      sample: 'var a = push ${x}',
+      call: function(v, x) { v.push(x); return v }
```

Comment:
I have no idea if the `return` changes the behavior compared to before this PR. The old code only did the `push`. Same for `unshift`.